api/python/t4/snapshots.py

```diff
@@ -333,18 +378,28 @@ def get(self, logical_key):
             when deserialization metadata is not present
         """
         entry = self._data[logical_key]
+
+        target_str = entry.meta.get('target')
+        if target_str is None:
+            raise HeliumException("No serialization metadata")
+
+        try:
+            target = TargetType(target_str)
+        except ValueError:
+            raise HeliumException("Unknown serialization target: %r" % target_str)
+
         physical_keys = entry.physical_keys
         if len(physical_keys) > 1:
             raise NotImplementedError
         physical_key = physical_keys[0] # TODO: support multiple physical keys
+
+        data = read_physical_key(physical_key)
+
         # TODO: verify hash
-        if 'target' in entry.meta:
-            # TODO: dispatch on target to deserialize
-            raise NotImplementedError
 
-        raise NotImplementedError
+        return deserialize_obj(data, target), entry.meta.get('user_meta')
 
-    def get_files(self, logical_key, path):
+    def copy(self, logical_key, dest):
```

Comment:
does `api.py` need to change with this? i.e. are we ready to make `get_file` and `put_file` into `copy`? OK to punt on this.

Reply:
Yeah, we probably want to do that - but wanted to make sure that URIs actually make sense